### ISSUE TITLE
Return existing state from AggregateStreamAsync when no events to apply to it

### ISSUE THREAD

**Issue by stevemossquill** [JasperFx/marten#2541]
I'm using [QueryEventStore.AggregateStreamAsync](https://github.com/JasperFx/marten/blob/master/src/Marten/Events/QueryEventStore.cs#L137) to implement [Aggregating Events into Existing State](https://martendb.io/events/projections/live-aggregates.html#aggregating-events-into-existing-state).

In the case where you have a starting state (aggregate), which newer events are applied to, could I propose a change to the logic, or possibly a new overload, please?

Currently the method has:

```c#
    public async Task<T?> AggregateStreamAsync<T>(Guid streamId, long version = 0, DateTimeOffset? timestamp = null,
        T? state = null, long fromVersion = 0, CancellationToken token = default) where T : class
    {
        var events = await FetchStreamAsync(streamId, version, timestamp, fromVersion, token).ConfigureAwait(false);
        if (!events.Any())
        {
            return null;
        }
```
 
The problem this gives me, is if `state` (the aggregate) is at the current version, `!events.Any()` will return `true` and return `null` from the method.

Whereas the behaviour I would like would be:

```c#
    public async Task<T?> AggregateStreamAsync<T>(Guid streamId, long version = 0, DateTimeOffset? timestamp = null,
        T? state = null, long fromVersion = 0, CancellationToken token = default) where T : class
    {
        var events = await FetchStreamAsync(streamId, version, timestamp, fromVersion, token).ConfigureAwait(false);
        if (!events.Any())
        {
            return state is null ? null : state;
        }
```

In other words, if there are no events to apply to `state` then just return `state` as is.

Without this, I have to first make a call to `session.Events.FetchStreamStateAsync` to find out the current version, which means an additional request to the database every time the aggregate is fetched.

**Comment by oskardudycz** [comment]
That's fair I updated that in #2543.

This change will be a step ahead to prepare us for archive scenarios when the stream is partially archived, and an empty stream might mean no existing one.

**Comment by oskardudycz** [comment]
@stevemossquill, it's released in `v6.0.0-rc.1` 🙂

**Comment by stevemossquill** [comment]
> @stevemossquill, it's released in `v6.0.0-rc.1` 🙂

Excellent. Many thanks for the quick response.
